tests/components/incomfort/test_water_heater.py

```diff
@@ -23,3 +26,26 @@ async def test_setup_platform(
     """Test the incomfort entities are set up correctly."""
     await hass.config_entries.async_setup(mock_config_entry.entry_id)
     await snapshot_platform(hass, entity_registry, snapshot, mock_config_entry.entry_id)
+
+
+@pytest.mark.parametrize(
+    "mock_heater_status",
+    [
+        MOCK_HEATER_STATUS | {"is_tapping": True},
+        MOCK_HEATER_STATUS | {"is_pumping": True},
+        MOCK_HEATER_STATUS | {"heater_temp": None},
+        MOCK_HEATER_STATUS | {"tap_temp": None},
+        MOCK_HEATER_STATUS | {"heater_temp": None, "tap_temp": None},
+    ],
+)
+@patch("homeassistant.components.incomfort.PLATFORMS", [Platform.WATER_HEATER])
+async def test_current_temperature_cases(
+    hass: HomeAssistant,
+    mock_incomfort: MagicMock,
+    entity_registry: er.EntityRegistry,
+    snapshot: SnapshotAssertion,
+    mock_config_entry: ConfigEntry,
+) -> None:
+    """Test incomfort entities with alternate current temperature calculation."""
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await snapshot_platform(hass, entity_registry, snapshot, mock_config_entry.entry_id)
```

Comment:
Maybe test the expected current temperature instead of doing a full snapshot (since there is already a snapshot of the water heater)?

Reply:
Changed